data/functions/utils.sql

```diff
@@ -5,6 +5,28 @@ begin
 end
 $$ language plpgsql immutable;
 
+-- Figures out the appropriate year to use for the transaction of a Schedule A
+-- or Schedule B record.  This function is used to fill in the value of the
+-- "transaction_two_year_period" column for the openFEC Schedule A and B tables
+-- and for checking how records should be split for the partitioning of those
+-- tables.  We are splitting by two-year cycles and want to make sure we only
+-- create tables for the cycles, not for every year present in the data.
+
+-- Params:
+--   transaction_date:  the value of the column representing the actual
+--                      transaction date for a given record.
+--   report_year:       the value of the column to use if the transaction_date
+--                      value is NULL.
+
+-- Returns:
+--   The calculated year to use as the transaction date of a record.
+create or replace function get_transaction_year(transaction_date date, report_year numeric)
+returns integer as $$
+begin
+    return coalesce(extract(year from transaction_date), report_year) + (coalesce(cast(extract(year from transaction_date) as numeric(4, 0)), report_year) % 2);
```

Comment:
IMO this would be more readable if the `coalesce(...)` were extracted into a variable--or, even better, if we used the existing `get_cycle` helper: https://github.com/18F/openFEC/blob/develop/data/functions/utils.sql#L1-L6